node/subsystem/src/lib.rs

```diff
@@ -38,13 +39,40 @@ pub mod util;
 #[cfg(any(test, feature = "test-helpers"))]
 pub mod test_helpers;
 
+/// How many slots are stack-reserved for active leaves updates
+///
+/// If there are fewer than this number of slots, then we've wasted some stack space.
+/// If there are greater than this number of slots, then we fall back to a heap vector.
+const ACTIVE_LEAVES_SMALLVEC_CAPACITY: usize = 8;
```

Comment:
Could maybe go even smaller, although I doubt it will impact performance or memory in any meaningful way

Reply:
Note, that given vec optimizations like https://github.com/rust-lang/rust/pull/72227/files, using smallvec will only save us one allocation and indirection for small active sets. The hardest part is to come with a benchmark to justify it.

Reply:
Agree that the right way to tune this is with a benchmark. Agree that it would be hard to write a really good benchmark for this. I think we all agree that 8 is a plausible size here anyway? For now, I think I'd prefer to just leave it than to spend the effort to determine the actual optimal size.